vsintegration/src/FSharp.Editor/Telemetry/TelemetryReporter.fs

```diff
@@ -89,29 +95,47 @@ type TelemetryReporter private (name: string, props: (string * obj) array, stopw
 
     static member val private SendAdditionalTelemetry =
         lazy
-            (let componentModel =
-                Package.GetGlobalService(typeof<ComponentModelHost.SComponentModel>) :?> ComponentModelHost.IComponentModel
-
-             if componentModel = null then
-                 TelemetryService.DefaultSession.IsUserMicrosoftInternal
-             else
-                 let workspace = componentModel.GetService<VisualStudioWorkspace>()
-                 workspace.Services.GetService<EditorOptions>().Advanced.SendAdditionalTelemetry)
+            (
+                let componentModel =
+                    Package.GetGlobalService(typeof<ComponentModelHost.SComponentModel>) :?> ComponentModelHost.IComponentModel
+
+                if isNull componentModel then
+                    TelemetryService.DefaultSession.IsUserMicrosoftInternal
+                else
+                    let workspace = componentModel.GetService<VisualStudioWorkspace>()
+
+                    TelemetryService.DefaultSession.IsUserMicrosoftInternal 
+                    || workspace.Services.GetService<EditorOptions>().Advanced.SendAdditionalTelemetry)
```

Comment:
Is this safe? Can it return null?

Reply:
it can, it means something is terribly wrong, and we need to crash